py/desispec/quicklook/procalgs.py

```diff
@@ -6,10 +6,40 @@
 import os,sys
 from desispec.quicklook import pas
 from desispec.quicklook import qlexceptions,qllogger
+from desispec.resolution import Resolution
 
 qlog=qllogger.QLLogger("QuickLook",20)
 log=qlog.getlog()
 
+def get_resolution(wave,flux,ivar,psf,usepsfboot=True):
+    """
+    Calculates approximage resolution values in the format that can directly
+    feed resolution data of desispec.frame.Frame object. 
+    
+    To zeroth order, we use psfboot xsigma values (constant resolution per fiber). 
+    Note: This is not the resolution of boxcar extraction!
+     
+    TODO: Replace this resolution to account for variation in dispersion direction
+          using extraction of arc and propagating the coefficients of the fit 
+          to possibly a new psf file so that resolution data can be evaluated on the fly 
+          for science exposures. This work is in progress.
+
+    wave: wavelength array
+    flux: (nspec,nwave) array of fluxes
+    ivar: (nspec,nwave) array of inverse variances
+    psf: desispec.psf.PSF like object
+    """
+    nspec=flux.shape[0]
+    nwave=flux.shape[1]
+    resolution_data=np.zeros((nspec,21,nwave))
+    if usepsfboot:
+        if hasattr(psf,'xsigma_boot'): #- only use if xsigma comes from psfboot
+            log.info("Getting resolution matrix band diagonal elements from constant Gaussing Xsigma")
+            for ispec in range(nspec):
+                thissigma=psf.xsigma(ispec,wave) 
+                Rsig=Resolution(thissigma)
+                resolution_data[ispec]=Rsig.data
+    return resolution_data
```

Comment:
Hi @sbailey Yes this is intended here. The reason is that xsigma is meant to be more general (may be calculated from different ways) but is equal to xsigma_boot when the psf used is coming from psfboot.

I have added a function to desispec.psf.PSF in the same commit, 

def xsigma(self,ispec,wave):
+        if hasattr(self,'xsigma_boot'):
+            if np.isscalar(wave):
+                return self.xsigma_boot[ispec]
+            else:
+                return np.full(len(wave),self.xsigma_boot[ispec]) #- constant xsigma for a given fiber

For now this is only taking the case where per fiber xsigma comes from psfboot and making copies when the wavelength is a vector, but this function is intended to be more general for case when there may/will be different approach to measure xsigma. This way the same function is sufficient given the content in psf object (coming from psfboot or otherwise). If this is conflicting, please let me know and I'll only have xsigma_boot explicitly for now and worry about xsigma when we really have a different approach for it.